### PR TITLE
fuse attention mask

### PR DESCRIPTION
**Pull request by williamberman**
re: https://github.com/huggingface/diffusers/issues/1891

I ran the unCLIP integration tests as well and verified they still passed. Thanks for the catch @Birch-san!

### CODE REVIEW

File: src/diffusers/models/cross_attention.py [hunk 1]
```diff
@@ -228,11 +232,12 @@ def __call__(self, attn: CrossAttention, hidden_states, encoder_hidden_states=No
         attention_mask = attn.prepare_attention_mask(attention_mask, sequence_length)
 
         query = attn.to_q(hidden_states)
-        query = attn.head_to_batch_dim(query)
 
         encoder_hidden_states = encoder_hidden_states if encoder_hidden_states is not None else hidden_states
         key = attn.to_k(encoder_hidden_states)
         value = attn.to_v(encoder_hidden_states)
+
+        query = attn.head_to_batch_dim(query)
```
Comment (williamberman) [inline src/diffusers/models/cross_attention.py:240]: nit: same statement order as `XFormersCrossAttnProcessor` and m'thinks more readable

File: src/diffusers/models/cross_attention.py [hunk 2]
```diff
@@ -185,17 +185,21 @@ def get_attention_scores(self, query, key, attention_mask=None):
...
+        else:
+            baddbmm_input = attention_mask
+
         attention_scores = torch.baddbmm(
-            torch.empty(query.shape[0], query.shape[1], key.shape[1], dtype=query.dtype, device=query.device),
+            baddbmm_input,
             query,
             key.transpose(-1, -2),
-            beta=0,
+            beta=1,
```
Comment (Birch-san) [inline src/diffusers/models/cross_attention.py:199]: `beta` being 0 when `attention_mask is None` had a special meaning: it skips addition altogether. the input tensor is unused. this saves us any cost of allocation.

Comment (williamberman) [reply src/diffusers/models/cross_attention.py:199]: ah, thanks!

Comment (williamberman) [reply src/diffusers/models/cross_attention.py]: fixed here https://github.com/huggingface/diffusers/pull/2111/commits/ab1a42d05f1898acf3e4004164e5b234bbe3fa21

**Review by patrickvonplaten — APPROVED** [review]
Looks good to me! Let's make sure we run the slow UNCLIP tests to make sure nothing is broken.

**Review by patil-suraj — APPROVED** [review]
Looks good to me!

**Comment by williamberman** [comment]
> Looks good to me! Let's make sure we run the slow UNCLIP tests to make sure nothing is broken.

Yep ran both slow and nightly unclip tests
